circle.yml

```diff
@@ -208,9 +262,30 @@ workflows:
           requires:
             - integration_tests_group_1
             - integration_tests_group_2
-  frontend-build-test:
+
+  build-frontend-test-and-deploy:
     jobs:
-      - build-frontend
+      - test-frontend
+      - build-frontend:
+          requires:
+            - test-frontend
+      # Frontend deployment (only on master branch)
+      - allow-deployment:
+          type: approval
+          requires:
+            - build-frontend
+          filters:
+            branches:
+              only:
+                - giovanni/add-frontend-ci-deployment
+      - deploy-frontend-stage:
+          requires:
+            - allow-deployment
```

Comment:
Do we really want to restrict the stage deployments with a click? It might be worth doing this for the prod deployment, but do we actually need this for stage?

Reply:
Isn't it better to wait the stage backend deployment before frontend changes?

Reply:
Anyway, it's not a big deal for stage, so I've removed it.

Reply:
> Isn't it better to wait the stage backend deployment before frontend changes?

Technically yes, but it doesn't matter for stage. My understanding of the CD for stage is that I merge the PR and I have the results automatically deployed. Adding a manual step that doesn't ping you when it's ready to be triggered (I checked their docs community forums, but it doesn't seem to be possible to trigger custom notification) makes it very easy to forget about it. At least I'm pretty sure I would forget to click it from time to time :D